ui/app/selectors/permissions.js

```diff
@@ -194,3 +194,50 @@ function getAccountsCaveatFromPermission (accountsPermission = {}) {
 function domainSelector (state, origin) {
   return origin && state.metamask.domains?.[origin]
 }
+
+export function getAccountToConnectToActiveTab (state) {
+  const selectedAddress = getSelectedAddress(state)
+  const connectedAccounts = getConnectedAccountsForActiveTab(state)
+
+  const { metamask: { identities } } = state
+  const numberOfAccounts = Object.keys(identities).length
+
+  if (connectedAccounts.length && connectedAccounts.length !== numberOfAccounts) {
+    if (connectedAccounts.findIndex(({ address }) => address === selectedAddress) === -1) {
+      return identities[selectedAddress]
+    }
+  }
+
+  return undefined
+}
+
+export function getConnectedAccountsForActiveTab (state) {
```

Comment:
Nit: This sounds like a synonym of `getPermittedAccountsForCurrentTab`. Maybe you could add `sorted` or `ordered` in there somewhere? :thinking: 

```suggestion
export function getOrderedConnectedAccountsForActiveTab (state) {
```